processor/baggage/lib/opentelemetry/processor/baggage/baggage_span_processor.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'opentelemetry-api'
+
+module OpenTelemetry
+  module Processor
+    module Baggage
+      class BaggageSpanProcessor
+      # Called when a {Span} is started, if the {Span#recording?} returns true.
+      #
+      # The BaggageSpanProcessor reads key/values stored in Baggage in the
+      # starting span's parent context and adds them as attributes to the span.
+      #
+      # Keys and values added to Baggage will appear on all subsequent child spans
+      # for a trace within this service *and* will be propagated to external services
+      # via propagation headers. If the external services also have a Baggage span
+      # processor, the keys and values will appear in those child spans as well.
+      #
+      # ⚠ ⚠ ⚠️
+      # To repeat: a consequence of adding data to Baggage is that the keys and
+      # values will appear in all outgoing HTTP headers from the application.
+      # Do not put sensitive information in Baggage.
+      # ⚠ ⚠ ⚠️
+      #
+      # @example
+      #   OpenTelemetry::SDK.configure do |c|
+      #     # Add the BaggageSpanProcessor to the collection of span processors
+      #     c.add_span_processor(Honeycomb::OpenTelemetry::Trace::BaggageSpanProcessor.new)
```

Comment:
```suggestion
      #     c.add_span_processor(OpenTelemetry::Processor::Baggage::BaggageSpanProcessor.new)
```

... depending on how the folder structure and module namespace for this processor shakes out her in contrib.